DigitalLearningSolutions.Data/DataServices/UserDataService.cs

```diff
@@ -105,6 +108,13 @@ public UserDataService(IDbConnection connection)
             return user;
         }
 
+        public DelegateUser? GetDelegateUserByIdFromCentre(int delegateId, int centreId)
+        {
+            var user = GetDelegateUserById(delegateId);
+
+            return user?.CentreId == centreId ? user : null;
+        }

```

Comment:
This feels like the logic that should live in a service rather than a data service/repo